app/services/sign_in/assertion_validator.rb

```diff
@@ -71,7 +71,7 @@ def validate_expiration
     end
 
     def create_new_access_token
```

Comment:
I have one more recommendation that I think will make things just a bit clearer. If we redefine `service_account_id`, then you don't have to do these `service_account_id || issuer` here and in `service_account_config`:

```
    def service_account_id
      @service_account_id ||= decoded_assertion_without_validation.service_account_id || decoded_assertion_without_validation.iss
    end
```